cmake/onnxruntime_common.cmake

```diff
@@ -185,18 +167,11 @@ if(MSVC)
   elseif(onnxruntime_target_platform STREQUAL "x86")
     set(X86 TRUE)
   endif()
-elseif(NOT onnxruntime_BUILD_WEBASSEMBLY)
-  if (CMAKE_OSX_ARCHITECTURES STREQUAL "arm64")
-    set(ARM64 TRUE)
-  elseif (CMAKE_OSX_ARCHITECTURES STREQUAL "arm64e")
-    set(ARM64 TRUE)
-  elseif (CMAKE_OSX_ARCHITECTURES STREQUAL "arm")
-    set(ARM TRUE)
-  elseif (CMAKE_OSX_ARCHITECTURES STREQUAL "x86_64")
-    set(X86_64 TRUE)
-  elseif (CMAKE_OSX_ARCHITECTURES STREQUAL "i386")
-    set(X86 TRUE)
+elseif(APPLE)
+  if(CMAKE_OSX_ARCHITECTURES_LEN LESS_EQUAL 1)
+    set(X64 TRUE)
```

Comment:
Is this right? What if I specify is to arm64 only, then x64 is true?

Reply:
Yes, it is only used at line 206 this file.

Reply:
This `x64` name is confusing